File: src/com/colacelli/irclib/connection/listeners/OnPingListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.connection.Connection;

public interface OnPingListener extends Listener {
    Type TYPE = Type.PING;

    void onPing(Connection connection);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/main/kotlin/com/colacelli/irclib/connection/listeners/OnPingListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.connection.Connection

interface OnPingListener : Listener {
    fun onPing(connection: Connection)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.PING
    }
}
